cli/pcluster/cfnconfig.py

```diff
@@ -239,81 +335,134 @@ def __init__(self, args):  # noqa: C901 FIXME!!!
+                        vpc_options.get(key)[1],
                         __temp__,
                     )
-                self.parameters[self.__vpc_options.get(key)[0]] = __temp__
+                self.parameters[vpc_options.get(key)[0]] = __temp__
             except configparser.NoOptionError:
                 pass
             except configparser.NoSectionError:
                 print(
                     "ERROR: VPC section [%s] used in [%s] section is not defined"
-                    % (self.__vpc_section, self.__cluster_section)
+                    % (vpc_section, self.__cluster_section)
                 )
                 sys.exit(1)
 
-        if __config.has_option(self.__cluster_section, "scheduler"):
-            self.parameters["Scheduler"] = __config.get(self.__cluster_section, "scheduler")
+    def __init_scheduler_parameters(self):
+        """
+        Validate scheduler related configuration settings and initialize corresponding parameters.
+        """
+        # use sge as default scheduler
+        if self.__config.has_option(self.__cluster_section, "scheduler"):
+            self.parameters["Scheduler"] = self.__config.get(self.__cluster_section, "scheduler")
         else:
             self.parameters["Scheduler"] = "sge"
 
-        # Validate region for batch
+        # check for the scheduler since AWS Batch requires different configuration parameters
         if self.parameters["Scheduler"] == "awsbatch":
-            self.__run_batch_validation(__config)
+            self.__init_batch_parameters()
         else:
-            # Set defaults outside the cloudformation template
-            self.parameters["MinSize"] = "0"
-            self.parameters["DesiredSize"] = "2"
-            self.parameters["MaxSize"] = "10"
-            for key in self.size_parameters:
-                try:
-                    __temp__ = __config.get(self.__cluster_section, key)
-                    if not __temp__:
-                        print("ERROR: %s defined but not set in [%s] section" % (key, self.__cluster_section))
-                        sys.exit(1)
-                    if key == "initial_queue_size":
-                        self.parameters["DesiredSize"] = __temp__
-                    elif key == "maintain_initial_size":
-                        self.parameters["MinSize"] = self.parameters.get("DesiredSize") if __temp__ == "true" else "0"
-                    elif key == "max_queue_size":
-                        self.parameters["MaxSize"] = __temp__
-                except configparser.NoOptionError:
-                    pass
+            self.__init_size_parameters()
+
+    def __init_size_parameters(self):
+        """
+        Initialize size parameters.
+        """
+        # Set defaults outside the cloudformation template
+        self.parameters["MinSize"] = "0"
+        self.parameters["DesiredSize"] = "2"
+        self.parameters["MaxSize"] = "10"
+
+        size_parameters = OrderedDict(
```

Comment:
Actually no, because the code assumes a precise order and the `self.parameters.get("DesiredSize")` must be already set when checking for the `maintain_initial_size`.
Probably we can refactor the code but I'd keep as is for now.